internal/gpustats/nvidia.go

```diff
@@ -417,21 +419,42 @@ func GetNvidiaGPUStatus() (NvidiaSmiLog, error) {
 	return ParseNvidiaSmi(output)
 }
 
-// Dummy struct to act as our adapter
-type NvidiaGPUHandler struct{}
+// Struct to act as our adapter
+type NvidiaGPUHandler struct {
+	lookup procinfo.UidLookup
+}
+
+func populateNames(samples *[]uplink.GPUStatSample, lookup procinfo.UidLookup) {
+	for i, sample := range *samples {
+		for j, proc := range sample.RunningProcesses {
+			name, err := lookup.Get(proc.Pid)
+			// Intentionally leave unresolved names as zero, don't want to fail
```

Comment:
Not failing is fine, but could we possibly still log the error so we're aware issues are occurring?

Reply:
then we would need to thread the needle with getting a logger down to this level. I dont think that is worth it. There might be other solutions but I rather not.

Reply:
Fair enough, feel free to merge without.
Has everyone else been using logger objects and passing them around? I've mainly just been using `slog.Info(...)` functions straight from the package

Reply:
hmmm, you have a point there. I've been mostly using logger objects

Reply:
> Has everyone else been using logger objects and passing them around?

It's super usefull on the control, as it allows associating log entries with the request that resulted in them.

Not sure if it matters here as much on satellite,